pkg/prometheus/promcfg.go

```diff
@@ -607,6 +607,11 @@ func (cg *ConfigGenerator) Generate(
 		Value: append(scrapeConfigs, addlScrapeConfigs...),
 	})
 
```

Comment:
Instead of checking inside method if we add check here we can avoid the method invocation itself
 `if p.Spec.EnableFeatures != nil && p.Spec.Exemplars != nil `

Reply:
done :)

Reply:
I'd be ok without the additional check here as long as appendStorageSettingsConfig() exits early if it needs to do nothing. The additional invocation isn't problematic since it's not in the "hot" path ;)

Reply:
I moved the check to the function `appendStorageSettingsConfig()`